### PR TITLE
Bryan: Created settings menu, light/dark mode, started navbar

### PR DESCRIPTION
**Pull request by bz5989**
Screenshot 2024-06-25 at 1 17 54 PM
Added in a button on top-right corner called settings. Inside, currently have a modified version of the light/dark button provided in components (as a separate file). Also includes minor nit changes from previous pull request

L=ENG-28
L=ENG-29

### CODE REVIEW

**Comment by bz5989** [comment]
Screenshot 2024-06-25 at 1 17 54 PM

File: js/src/personal/bryan/LightDark.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Button, useMantineColorScheme } from '@mantine/core'
+
+export function LightDark() {
```
Comment (arklian) [inline js/src/personal/bryan/LightDark.tsx:3]: We likely should start adding JSDoc to comment on what the component is supposed to do.
`/** Component to toggle the light or dark mode of the page. */`
Or something of the sort.

File: js/src/personal/bryan/TimeDisplay.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ export function TimeDisplay() {
     'Dec',
   ]
   const [date, setDate] = useState(new Date())
-
+  // useEffect is used here to allow setsDate to rerun every second
```
Comment (arklian) [inline js/src/personal/bryan/TimeDisplay.tsx:21]: In comments, you generally want to be putting the "Why" and not the "How".

Your comment directly mirrors the code and can be read from the code.
Something like
` // Refresh the time every second to keep it at the current time.`
tells you why the useEffect exists.

**Comment by bz5989** [comment]
New example image
Screenshot 2024-06-25 at 2 21 52 PM

File: js/src/personal/bryan/LightDark.tsx (hunk 1 above)
Comment (arklian) [reply js/src/personal/bryan/LightDark.tsx:4]: This comment should go above `export function LightDark` and not above the imports. Same in other files.

Some IDEs collapse imports by default, and this comment applies more to the component itself than the whole file.

**Review by arklian — APPROVED** [review]

**Pull request merged**